src/analysis/processing/qgsalgorithmzonalhistogram.cpp

```diff
@@ -94,6 +109,12 @@ QVariantMap QgsZonalHistogramAlgorithm::processAlgorithm( const QVariantMap &par
   if ( !zones )
     throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT_VECTOR" ) ) );
 
+  if ( !mIsInteger )
```

Comment:
I'd move that block into prepareAlgorithm() which would remove the need for the mIsInteger member now that we're just warning.,